lib/app/builder.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class Builder
```

Comment:
A little bit complicated.
Nit: read about rails validations. Easy way to do validation logic